java/Makefile

```diff
@@ -1168,13 +1279,20 @@ DAIKON_PACKAGES := $(shell ls -F daikon | grep / | egrep -v 'util' | egrep -v 'd
 .PHONY: javadoc-files.txt
 javadoc-files.txt:
 	$(MAKE) ${AUTO_GENERATED_FILES}
-	find * \( -name 'PrototypeChecker.java' -o -name 'VIndexChecker.java' -o -name 'VIndexAnnotatedTypeFactory.java' -o -path 'daikon/dcomp-transfer' \) -prune -o -name "*.java" -print | ${SORT_DIRECTORY_ORDER} > javadoc-files.txt
+	find * \( -name 'PrototypeChecker.java' -o -name 'VIndexChecker.java' -o -name 'VIndexAnnotatedTypeFactory.java' -o -path 'daikon/dcomp-transfer' \) -prune -o -name "*24.java" -prune -o -name "*.java" -print | ${SORT_DIRECTORY_ORDER} > javadoc-files.txt
 
 # If Java 8, don't run javadoc; a bug in javadoc causes it to crash.
 ifeq ($(JAVA8), 1)
   SKIP_JAVADOC=1
 endif
 
+# When JDK 24 is offically released, remove '--enable-preview'
+ifeq ($(shell test ${JAVA_RELEASE_NUMBER} -gt 23; echo $$?),0)
```

Comment:
Can this use `${extra-error-prone-args-2}` (which in that case should be renamed)?  Or does javadoc not accept `-source 24`?